### PR TITLE
Make rhymes return rhymes for all pronunciations

### LINKED ISSUE #45: rhymes() returns only words rhyming with first pronunciation
The docs state:

> The `pronouncing.rhymes()` function returns a list of all possible rhymes for the given word—i.e., words that rhyme with any of the given word’s pronunciations. 

It appears, however, that the `pronouncing.rhymes()` function in fact returns a list of a list of words that rhyme with *only the first* of the given word's pronunciations.

### PR DESCRIPTION
**Pull request by jdbean**
Rhymes now returns a single, flat, sorted list that contains the words rhyming with all pronunciations of the given word.

Closes #45

### CODE REVIEW

File: pronouncing/__init__.py [hunk 1]
```diff
@@ -225,6 +225,7 @@ def rhymes(word):
                                     element), []) if w != word])
         combined_rhymes = list(chain.from_iterable(combined_rhymes))
         combined_rhymes.sort()
-        return combined_rhymes
+                unique_combined_rhymes = set(combined_rhymes)
```
Comment (hugovk) [inline pronouncing/__init__.py]: There's an unexpected indent here. Should it be inline with the others?

Comment (jdbean) [reply pronouncing/__init__.py]: Absolutely. Sorry about that. Should be taken care of now.

File: pronouncing/__init__.py [hunk 2]
```diff
@@ -217,8 +218,14 @@ def rhymes(word):
...
-        return [w for w in rhyme_lookup.get(rhyming_part(phones[0]), [])
-                if w != word]
+    combined_rhymes = []
+    if phones:
+        for element in phones:
+            combined_rhymes.append([w for w in rhyme_lookup.get(rhyming_part(
+                                    element), []) if w != word])
+        combined_rhymes = list(chain.from_iterable(combined_rhymes))
+        combined_rhymes.sort()
+        unique_combined_rhymes = set(combined_rhymes)
```
Comment (aparrish) [inline pronouncing/__init__.py]: iirc, I'm not sure if using set/list to remove duplicates will produce the desired results across all the versions of python that we're targeting—probably best to sort the list post-dedup.

**Comment by aparrish** [comment]
also this needs an update to the `test_rhymes` test (which is why the CI is currently failing). can you update that test and push to this same PR? (If you don't have time or prefer not to, that's fine—I can do it, but it might take a while for me to get around to it.)

**Comment by aparrish** [comment]
actually if you could add a test to avoid regressions on this that would be great!

**Comment by jdbean** [comment]
I think that should do it! :smile:

File: tests/test_pronouncing.py [hunk 3]
```diff
@@ -65,6 +65,16 @@ def test_rhymes(self):
...
+        # ensure correct behavior for words that have multiple pronunciations
+        rhymes = pronouncing.rhymes("dove")
+        expected = [
+            "above", "belove", "boeve", "bove", "clove", "cove",
+		    "deneuve", "drove", "glove", "gov", "gove", "grove",
+		    "hove", "labauve", "labov", "labove", "love", "nov",
+		    "o'glove", "of", "rove", "shove", "soave", "stove",
+            "strove", "thereof", "throve", "tov", "trove", "vanhove",
+		    "wove"]
+        self.assertEqual(expected, rhymes)
```
Comment (hugovk) [inline tests/test_pronouncing.py]: Looks good. 

Please could you add a newline between each commented block for readability?

(They could even be in their own test methods so only one thing is tested in each.)

**Comment by jdbean** [comment]
@hugovk Thanks for the feedback. I agree that the `rhymes` tests are better off being split into their own methods. I've pushed a new commit which splits them out as such.

**Comment by aparrish** [comment]
looks good, thanks! will include in next release.

**Comment by jdbean** [comment]
Most welcome. Glad I could help!

**Pull request merged**